### PR TITLE
Transfer URI Correctly

### PR DESCRIPTION
**Pull request by cssuh**
Before: Saving a query before executing would put the query editor into a disconnected state (because a query runner does not exist)

After: Connection URI is transferred correctly even without a query runner present

STS Changes: https://github.com/microsoft/sqltoolsservice/pull/2265

Fixes https://github.com/microsoft/azuredatastudio/issues/21988

### CODE REVIEW

File: src/sql/workbench/services/query/common/queryManagement.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/services/query/common/queryManagement.ts:343]: We probably shouldn't be changing the URI in this case since that could cause incorrect results for anything else using that URI. Better to fail out immediately IMO. Ideally we'd also want to clean up/disconnect the old query URI too, although that may need to be a follow up item to look in to if it ends up being a huge mess...

Comment (cssuh) [reply src/sql/workbench/services/query/common/queryManagement.ts:343]: ok, added the throw new Error part back in.

**Review by Charles-Gagnon — APPROVED** [review]
Whenever you update STS please always include the list of changes being brought in and tag the people making the commits so they're aware of the update.

(in this case though probably easiest to wait until Cory gets their PR in and then merge in to this since they've already done that work for a lot of the commits)

**Pull request merged**
